torch/_inductor/codegen/wrapper.py

```diff
@@ -1497,6 +1497,7 @@ def can_reuse(self, input_buffer, output_buffer=None):
             or name in V.graph.graph_inputs
             or name in V.graph.constants
             or name in V.graph.never_reuse_buffers
+            or name not in self.allocated
```

Comment:
had this weird case where we had back-to-back re-uses

```
buf11 = buf6; del buf6
buf12 = buf11; del buf11
```

Reply:
found out it was because we had ` V.kernel.inplace_update_buffers[buf12] = buf11`, although `buf11` is a `NopKernelSchedulerNode(MutationOutput)`.

That happens here.
https://github.com/pytorch/pytorch/blob/0feab7d6c3c7da284a86d6a0c7d0f0c84cff577e/torch/_inductor/scheduler.py#L404-L406

Reply:
I can't add this line, some buffers are not allocated but are "reused". Like the buffer for a `ir.WhileLoop`. It can store subgraph output buffers, and eventually be re-used by a buffer in the parent graph.

https://github.com/pytorch/pytorch/blob/a32eac345f5d7862246c397d0a32b5e0cbca77ff/torch/_inductor/codegen/wrapper.py#L1585-L1594

So, instead we skip inserting into `inplace_update_buffers` if `isinstance(input_node, NopKernelSchedulerNode)`.